lib/v2/screens/sign_up/claim_invite/components/bottom_container.dart

```diff
@@ -0,0 +1,178 @@
+                  style: Theme.of(context).accentTextTheme.subtitle2,
+                ),
+                const SizedBox(height: 16),
+                TextFormField(
+                  controller: _keyController,
+                  style: Theme.of(context).accentTextTheme.bodyText1,
+                  decoration: InputDecoration(
+                    hintText: 'Invite code (5 words)'.i18n,
+                    hintStyle: Theme.of(context).textTheme.subtitle2OpacityEmphasis,
+                    errorText: _currentState.pageState == PageState.failure ? _currentState.errorMessage : null,
+                    suffixIcon: _inviteCodeSuffixIcon,
+                    enabledBorder: const OutlineInputBorder(
+                      borderRadius: BorderRadius.all(Radius.circular(8)),
+                      borderSide: BorderSide(
+                        width: 1,
+                        color: Colors.black,
+                      ),
+                    ),
+                    focusedBorder: const OutlineInputBorder(
+                      borderRadius: BorderRadius.all(Radius.circular(8)),
+                      borderSide: BorderSide(
+                        width: 1,
+                        color: Colors.black,
+                      ),
+                    ),
+                    focusedErrorBorder: const OutlineInputBorder(
+                      borderRadius: BorderRadius.all(Radius.circular(8)),
+                      borderSide: BorderSide(
+                        width: 1,
+                        color: AppColors.red,
+                      ),
+                    ),
+                    errorBorder: const OutlineInputBorder(
+                      borderRadius: BorderRadius.all(Radius.circular(8)),
+                      borderSide: BorderSide(
+                        width: 1,
+                        color: AppColors.red,
+                      ),
+                    ),
+                  ),
+                ),
+                const SizedBox(height: 50),
+                FlatButtonLong(
+                  title: "Claim code".i18n,
+                  onPressed: _onClaimPressed(context),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget get _inviteCodeSuffixIcon {
+    final bool isChecked = _currentState.pageState == PageState.success && _currentState.inviteModel != null;
+    final bool canClear = !isChecked && _keyController.text.isNotEmpty;
+    final bool loading = _currentState.pageState == PageState.loading;
+
+    return QuadStateClipboardIconButton(
+      onClear: () {
+        setState(() {
+          _keyController.text = '';
+        });
+      },
+      onPaste: () async {
+        final clipboardText = await getClipboardData();
+        setState(() {
+          _keyController.text = clipboardText;
+        });
+      },
+      isChecked: isChecked,
+      canClear: canClear,
+      isLoading: loading,
+    );
+  }
```

Comment:
Please refactor _inviteCodeSuffixIcon  as stateless widget class no getters, no vars no funtions.
You can access the state here with Blocbuilder too.

Reply:
This is already a stateless widget but I moved the logic into the state and removed the getter.